nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,459 @@
+@SupportsBatching
+@Tags({"elasticsearch", "elasticsearch 5", "fetch", "read", "get"})
+@CapabilityDescription("Retrieves a document from Elasticsearch using the specified connection properties and the "
+        + "identifier of the document to retrieve. If the cluster has been configured for authorization and/or secure "
+        + "transport (SSL/TLS), and the X-Pack plugin is available, secure connections can be made. This processor "
+        + "supports Elasticsearch 5.x clusters.")
+public class JsonQueryElasticsearch5 extends AbstractProcessor {
+
+    private RestClient client;
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("All original flowfiles that don't cause an error to occur go to this relationship").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be read from Elasticsearch are routed to this relationship").build();
+
+    public static final Relationship REL_HITS = new Relationship.Builder().name("hits")
+            .description("Search hits are routed to this relationship.")
+            .build();
+
+    public static final Relationship REL_AGGREGATIONS = new Relationship.Builder().name("aggregations")
+            .description("Aggregations are routed to this relationship.")
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("el5-fetch-index")
+            .displayName("Index")
+            .description("The name of the index to read from")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("el5-type")
+            .displayName("Type")
+            .description("The type of this document (used by Elasticsearch for indexing and searching)")
+            .defaultValue("")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("el5-query")
+            .displayName("Query")
+            .description("A query in JSON syntax, not Lucene syntax.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor HTTP_HOSTS = new PropertyDescriptor.Builder()
+            .name("el5-http-hosts")
+            .displayName("HTTP Hosts")
+            .description("A comma-separated list of HTTP hosts that host ElasticSearch query nodes.")
+            .required(true)
+            .expressionLanguageSupported(false)
```

Comment:
Whether or not Expression Language has been historically supported in other ES processors, this seems like a good place to support them :) Pretty much goes for all these properties, except maybe sensitive properties (for now) like password. I realize a parameterized username might be useless without a parameterized password, but no harm no foul!

Reply:
I think other processors have allowed EL in the password, it's not very secure without encrypted properties/registry entries, but that might be acceptable for some use cases